docs/source/conf.py

```diff
@@ -44,24 +45,25 @@
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
 extensions = [
-    'sphinx.ext.autodoc',
-    'sphinx.ext.autosummary',
-    'sphinx.ext.doctest',
-    'sphinx.ext.ifconfig',
-    'sphinx.ext.mathjax',
-    'sphinx.ext.viewcode',
-    'sphinx.ext.napoleon',
-    'myst_parser',
-    'sphinx_reredirects',
+    "sphinx.ext.autodoc",
+    "sphinx.ext.autosummary",
+    "sphinx.ext.doctest",
+    "sphinx.ext.ifconfig",
+    "sphinx.ext.mathjax",
+    "sphinx.ext.viewcode",
+    "sphinx.ext.napoleon",
+    "myst_parser",
+    "sphinx_reredirects",
+    "rustdoc_trim",
```

Comment:
Here we register `rustdoc_trim` so it'll run during Sphinx build, meaning we don't need to manually fix it in source files that are in a temp dir, allowing us to use original build dir for simplicity.

Tested by checking the catalog page, can see it's properly trimmed:

<img width="796" height="695" alt="Image" src="https://github.com/user-attachments/assets/2d9f8133-644a-4b2d-95bc-922307818703" />

Source code:

https://github.com/apache/datafusion/blob/a45192084e95a2d629b7cbf88b43133d8d6fb814/docs/source/library-user-guide/catalogs.md?plain=1#L56-L68